### PR TITLE
chore: update wgpu and realign to spec

### LINKED ISSUE #9664: Strange interaction on debug builds + Mac + WebGPU + ICU data
On my Mac (x86_64-apple-darwin) at commit 7cd14f97c or later on a debug build, I get segfaults from `deno`.  Release builds do not exhibit the behaviour, but it does mean that I can't run tests locally.  I can most concisely recreate it using the REPL:

```
$ deno
Deno 1.7.4
exit using ctrl+d or close()
> Intl.Collator("en-us")
zsh: segmentation fault  deno
```

I tried changing the alignment of the ICU data in core to `Align32` and `Align64` but it did not change the outcome.

### LINKED ISSUE #9686: Building Alpine linux now requires libc++ (and then prints warnings)
Running deno executable on alpine (frolvlad/alpine-glibc since glibc is required) fails since 1.8.0:

```
# running deno
deno: error while loading shared libraries: libstdc++.so.6: cannot open shared object file: No such file or directory
```
if I install libstdc++ via `apk add libstdc++` I see (when running deno) a warning repeated 5 times:

```
deno: /usr/lib/libstdc++.so.6: no version information available (required by deno)
deno: /usr/lib/libstdc++.so.6: no version information available (required by deno)
deno: /usr/lib/libstdc++.so.6: no version information available (required by deno)
deno: /usr/lib/libstdc++.so.6: no version information available (required by deno)
deno: /usr/lib/libstdc++.so.6: no version information available (required by deno)
Download https://deno.land/std/examples/welcome.ts
Warning Implicitly using latest version (0.89.0) for http://deno.land/std/examples/welcome.ts
Download https://deno.land/std@0.89.0/examples/welcome.ts
Check https://deno.land/std/examples/welcome.ts
Welcome to Deno!
```

This is a regression since previously `libstdc++` was not required to be dynamically linked. As well as the warning being somewhat loud/unnecessary (?).

cc @lucacasonato @caspervonb 

https://github.com/hayd/deno-docker/runs/2024710911?check_suite_focus=true

### PR DESCRIPTION
**Pull request by lucacasonato**
Closes #9664
Closes #9686

Preliminary wgpu update patch to align to spec again.

### CODE REVIEW

**Comment by kitsonk** [comment]
@lucacasonato what is the status of this? Since we have landed some LSP features that always use the ICU data, I can't use debug builds of Deno anymore to develop with (because of #9664) and always have to build `--release` which is making things very difficult.

**Comment by crowlKats** [comment]
@kitsonk for that we are waiting on wgpu 0.8, which relies on https://github.com/gfx-rs/gfx/milestone/16, so i assume soon, as all the needed PRs got merged

**Comment by lucacasonato** [comment]
@crowlKats PTAL

File: extensions/webgpu/queue.rs [hunk 1]
```diff
@@ -136,10 +138,14 @@ pub fn op_webgpu_write_texture(
...
+  let data_layout = wgpu_types::ImageDataLayout {
     offset: args.data_layout.offset.unwrap_or(0),
-    bytes_per_row: args.data_layout.bytes_per_row.unwrap_or(0),
-    rows_per_image: args.data_layout.rows_per_image.unwrap_or(0),
+    // TODO(lucacasonato): check with spec if non zero is correct here
+    bytes_per_row: NonZeroU32::new(args.data_layout.bytes_per_row.unwrap_or(0)),
+    // TODO(lucacasonato): check with spec if non zero is correct here
+    rows_per_image: NonZeroU32::new(
+      args.data_layout.rows_per_image.unwrap_or(0),
+    ),
```
Comment (crowlKats) [inline extensions/webgpu/queue.rs:148]: same here

File: extensions/webgpu/shader.rs [hunk 2]
```diff
@@ -56,14 +56,9 @@ pub fn op_webgpu_create_shader_module(
...
 
-  let mut flags = wgpu_types::ShaderFlags::default();
-  flags.set(wgpu_types::ShaderFlags::VALIDATION, true);
-  #[cfg(all(target_os = "macos", target_arch = "x86_64"))]
-  flags.set(wgpu_types::ShaderFlags::EXPERIMENTAL_TRANSLATION, true);
-
   let descriptor = wgpu_core::pipeline::ShaderModuleDescriptor {
     label: args.label.map(Cow::from),
-    flags,
+    flags: wgpu_types::ShaderFlags::default(),
```
Comment (crowlKats) [inline extensions/webgpu/shader.rs:61]: Why disable shader validation?

Comment (lucacasonato) [reply extensions/webgpu/shader.rs:61]: Changed to `wgpu_types::ShaderFlags::all()`.

**Review by ry — APPROVED** [review]
LGTM

**Review by crowlKats — APPROVED** [review]
LGTM

**Pull request merged**